### DISCUSSION TITLE
Newbie requiring help Is this possible?

### DISCUSSION THREAD

**Discussion by bacardiiii** [kellerza/sunsynk#123] [General]
I have a Pi3 model b running solar assistant in the garage, it is connected to the rs485 via a usb adapter. I would like to get your addon working.

I have connected to solar assistant via mqtt bridge in home assistant (my home assistant is on another pi in the house). Is there anyway I can use mqtt and the pi in the garage? If I can’t, can I split the rs485 into two. and then run two usb rs485 adapters

**Comment by kellerza** [comment]
Solar assistant and this addon does the same thing when it comes to reporting inverter values to HA

why do you want to do this?

**Comment by kellerza** [comment]
The main difference would be that this addon allows you to set inverter parameters in HA. 

Not sure if SA provides that functionality yet. When I evaluated it you could already set options, but not using HA/MQTT

**Reply by bacardiiii** [reply]
Thank you for your reply 

I was looking through Facebook and saw someone had a better work mode layout. They had start and end times for grid charge (like on the inverter). I assumed this was your addon (apologies if I’ve got it wrong). 

I can use the Mqtt from solar assistant to set values much like the Sunsynk connect app. Hence why I was hoping to try your addon

**Reply by kellerza** [reply]
If you can get&set the values, you can do anything with lovelace layouts. No need for a new addon.

There are some examples in the addon’s docs to get you started. The home-assistant community is also great for questions on layouts

**Reply by bacardiiii** [reply]
I can’t  get a start and end time, like there is on the inverter just the start. Apart from that it’s fine.

thank you for your time

**Reply by kellerza** [reply]
The end time is the start time of the next program… the last wraps around to the first

**Comment by rebelchris** [comment]
I also have a related question.

My HA runs on a pi located on the other end of the house (compared to inverter) 
So can't directly plug a cable into that one.

I assume the modbus would be best scenario then?
It simply runs off a secondary pi and transmits data to the HA pi?
Wasn't quite sure if this was intended from the documentation and how to setup this secondary pi.

**Reply by kellerza** [reply]
Not sure I understand your question - either option 2a/b/c or 3 from
https://kellerza.github.io/sunsynk/guide/deployment-options

PRs are always welcome to help clarify the docs or redraw the diagrams… you also have to be realistic though, the intention of the docs is not to help you setup Home Assistant OS. There are 100s of videos/1000s of docs on that

**Reply by rebelchris** [reply]
Got it, was indeed referring to option 2, but wasn't quite sure how that installation would work because it's outside of HA.

**Reply by kellerza** [reply]
- 2a is also home assistant
- 2b is linux
- 2c is a dedicated device like https://kellerza.github.io/sunsynk/guide/deployment-options#rs485-gateways
